src/front/app/components/ChallengeTopics/index.js

```diff
@@ -0,0 +1,49 @@
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { getChallengeTopics, toggleChallengeTopic } from './../../actions/challengeTopics';
+import ChallengeTopic from './ChallengeTopic';
+
+class ChallengeTopicList extends Component {
+    constructor(props) {
+        super(props);
+        this.handleClick = this.handleClick.bind(this);
+    }
+
+    componentDidMount() {
+        const { dispatch } = this.props;
+        dispatch(getChallengeTopics());
+    }
+
+    handleClick(name) {
+        const { dispatch } = this.props;
+
+        dispatch(toggleChallengeTopic(name));
+    }
+
+    render() {
+        const { challengeTopics } = this.props;
+
+        return (
+            <ul>
+                {challengeTopics.map((topic, index) => (
+                    <ChallengeTopic
+                      key={index}
+                      {...topic}
+                      onClick={() => {this.handleClick(topic.name);}}
```

Comment:
you shouldn`t call handler here, just pass handler to ChallengeTopic component and call it there with arguments you need